### PR TITLE
Update Transformer's chatbot tutorial not to conflict with ChatModel tutorials

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/13787?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13787/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13787
```

</p>
</details>

### What changes are proposed in this pull request?

[This tutorial page](https://mlflow.org/docs/latest/llms/transformers/tutorials/conversational/pyfunc-chat-model.html) shows up at the top rank when searching for "MLflow chat model" on Google. This is a bit old page and also primary talks about Transformers `llm/v1/chat` feature before `ChatModel`, so not an ideal place for users to visit for the guidance about `ChatModel`.

This PR changes the page and title to focus on Transformers only so it won't surpass the newer ChatModel tutorials on web search.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

**Review by daniellok-db — APPROVED** [review]
makes sense, i agree it's a bit confusing to have two different tutorials that reference ChatModel. thanks for this!

File: docs/source/llms/transformers/tutorials/conversational/pyfunc-chat-model.ipynb [hunk 1]
```diff
@@ -4,19 +4,22 @@
...
-    "## Building and Serving an OpenAI-compatible Chatbot\n",
+    "## Deploying a Transformer model as an OpenAI-compatible Chatbot\n",
     "\n",
-    "Welcome to our tutorial on using Transformers and MLflow to create an OpenAI-compatible chat model. In MLflow 2.11 and up, the [ChatModel](https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#mlflow.pyfunc.ChatModel) class has been added, allowing for convenient creation of served models that conform to the OpenAI API spec. This enables you to seamlessly swap out your chat app's backing LLM, or to easily evaluate different models without having to edit your client-side code.\n",
+    "Welcome to our tutorial on using Transformers and MLflow to create an OpenAI-compatible chat model. In MLflow 2.11 and up, MLflow's Transformers flavors support special task type `llm/v1/chat`, which turns thousands of [text-generation](https://huggingface.co/models?pipeline_tag=text-generation) model on Hugging Face into conversational chat bots interoperable with OpenAI models. This enables you to seamlessly swap out your chat app’s backing LLM, or to easily evaluate different models without having to edit your client-side code.\n",
```
Comment (BenWilson2) [inline docs/source/llms/transformers/tutorials/conversational/pyfunc-chat-model.ipynb:9]:
```suggestion
    "Welcome to our tutorial on using Transformers and MLflow to create an OpenAI-compatible chat model. In MLflow 2.11 and up, MLflow's Transformers flavors support special task type `llm/v1/chat`, which turns thousands of [text-generation](https://huggingface.co/models?pipeline_tag=text-generation) models on Hugging Face into conversational chat bots that are interoperable with OpenAI models. This enables you to seamlessly swap out your chat app’s backing LLM or to easily evaluate different models without having to edit your client-side code.\n",
```

File: docs/source/llms/transformers/tutorials/conversational/pyfunc-chat-model.ipynb [hunk 2]
```diff
@@ -4,19 +4,22 @@
...
     "\n",
     "If you haven't already seen it, you may find it helpful to go through our [introductory notebook on chat and Transformers](https://mlflow.org/docs/latest/llms/transformers/tutorials/conversational/conversational-model.html) before proceeding with this one, as this notebook is slightly higher-level and does not delve too deeply into the inner workings of Transformers or MLflow Tracking.\n",
     "\n",
+    "\n",
+    "**Note**: This page covers how to deploy a **Transformers** models as a chatbot. If you are using different framework or custom python model, use [ChatModel](https://mlflow.org/docs/latest/llms/chat-model-intro/index.html) instead to build an OpenAI-compatible chat bot.\n",
```
Comment (BenWilson2) [inline docs/source/llms/transformers/tutorials/conversational/pyfunc-chat-model.ipynb:14]:
```suggestion
    "**Note**: This page covers how to deploy a **Transformers** models as a chatbot. If you are using a different framework or a custom python model, use [ChatModel](https://mlflow.org/docs/latest/llms/chat-model-intro/index.html) instead to build an OpenAI-compatible chat bot.\n",
```

File: docs/source/llms/transformers/tutorials/conversational/pyfunc-chat-model.ipynb [hunk 3]
```diff
@@ -4,19 +4,22 @@
...
+    "\n",
     "### Learning objectives\n",
     "\n",
     "In this tutorial, you will:\n",
     "\n",
     "- Create an OpenAI-compatible chat model using TinyLLama-1.1B-Chat\n",
-    "- Serve the model with MLflow Model Serving\n",
-    "- Learn how to use MLflow's `pyfunc` API to add arbitrary customization to your model"
+    "- Log the model to MLflow and loaded it back for local inference.\n",
```
Comment (BenWilson2) [inline docs/source/llms/transformers/tutorials/conversational/pyfunc-chat-model.ipynb:21]:
```suggestion
    "- Log the model to MLflow and load it back for local inference.\n",
```

**Review by BenWilson2 — APPROVED** [review]
LGTM once nits are addressed

**Pull request merged**